task/source-build/0.1/source-build.yaml

```diff
@@ -0,0 +1,86 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: source-build
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "appstudio, hacbs"
+spec:
+  description: Source image build.
+  params:
+    - name: BINARY_IMAGE
+      description: Binary image name from which to generate the source image name.
+      type: string
+    - name: BASE_IMAGES
+      description: >-
+        Base images used to build the binary image. Each image per line in the same order of FROM
+        instructions specified in a multistage Dockerfile. Default to an empty string, which means
+        to skip handling a base image.
+      type: string
+      default: ""
+  results:
+    - name: BUILD_RESULT
+      description: Build result.
+    - name: SOURCE_IMAGE_URL
+      description: The source image url.
+    - name: SOURCE_IMAGE_DIGEST
+      description: The source image digest.
+  workspaces:
+    - name: workspace
+      description: The workspace where source code is included.
+  volumes:
+    - name: source-build-work-place
+      emptyDir:
+        medium: Memory
```

Comment:
Are you sure about memory volume? What's the average estimated consumption?

Reply:
in osbs source can have up to 20-40gb

Reply:
but I certainly wouldn't be adding sources in memory and working with that

Reply:
> Are you sure about memory volume? What's the average estimated consumption?

We don't have real source image build yet. The existing general idea comes from the OSBS that is source image build would consume storage too much as we mentioned multiple times in the meetings.

> in osbs source can use up to 20-40gb (but that was including srpms)

@mmorhun Does it work if we set PVC to request 20Gi for the task for every build pipeline?